src/tup/parser.c

```diff
@@ -1130,7 +1129,7 @@ int import(struct tupfile *tf, const char *cmdline, const char **retvar, const c
 /* If a .gitignore directive is removed, we need to either revert back to the
  * user's explicit .gitignore file, or remove it entirely.
  */
-static int remove_tup_gitignore(struct tupfile *tf, struct tup_entry *tent)
+int remove_tup_gitignore(struct graph *g, struct tup_entry *tent)
```

Comment:
This should probably still take a struct tupfile, since this function should be printing errors to `tf->f` instead of `stderr`. The former will print the error messages under the banner of the directory being parsed, rather than immediately, so it is more obvious which directory actually failed. I can fix this in a separate commit if you like.

Reply:
Err in retrospect, keeping the struct tupfile would make moving it into updater.c more difficult. I am making a change now to make this a FILE *err input instead, so the updater.c usage of it can pass in stderr and the parser.c usage can pass in tf->f.

Reply:
Ah, yeah, I was wondering if we still had a valid tupfile in updater.c for that :sweat_smile:

Reply:
Alright, I left this one for you, since you said you already had a patch waiting; but I fixed the other issues. I think it might be good to go, but you'll be the judge of that XD